src/ifcopenshell-python/ifcopenshell/type_checker.py

```diff
@@ -0,0 +1,149 @@
+from lark import Lark, Transformer
+
+# Grammar to retrieve the following structure: YYYY-MM-DDThh:mm:ss
+grammar = """
+    start: time_value
+    time_value: year "-" month "-" day "T" hour ":" minute ":" second ["." millisecond]
+
+    hour: DIGIT DIGIT
+    minute: DIGIT DIGIT
+    second: DIGIT DIGIT
+    millisecond: DIGIT+
+
+    day: DIGIT DIGIT
+    month: DIGIT DIGIT
+    year: MINUS? DIGIT+
+
+    MINUS: "-"
+    DIGIT: "0".."9"
+
+    %import common.WS
+    %ignore WS
+"""
+
+parser = Lark(grammar, start='time_value')
```

Comment:
`start='time_value'` still needed? You already defined it in the grammar. Also suggest `datetime_value`.

Reply:
fixed in https://github.com/IfcOpenShell/IfcOpenShell/pull/3864/commits/9021d126ef28ed3687eee3c6eed0d48c8ce4d42c